engine/src/test/java/org/hibernate/search/engine/mapper/mapping/building/impl/ConfiguredIndexSchemaManagerNestingContextTest.java

```diff
@@ -921,10 +921,96 @@ public void indexedEmbedded_includePaths_embedding_depth2AndIncludePaths() {
 		// Same includePaths as above
 		checkSimpleIndexedEmbeddedExcluded(
 				level3Context, typeModel4Mock, "nested.",
-				2, includePaths
+				2, includePaths, null
 		);
 	}
 
+	@Test
+	@TestForIssue(jiraKey = "HSEARCH-1182")
+	public void indexedEmbedded_excludePaths() {
+		ConfiguredIndexSchemaNestingContext rootContext = ConfiguredIndexSchemaNestingContext.root();
+
+		Set<String> excludePaths = new HashSet<>();
+
+		excludePaths.add( "foo" );
+		excludePaths.add( "bar" );
+		excludePaths.add( "foo.bar" );
+
+		ConfiguredIndexSchemaNestingContext level1Context = checkSimpleIndexedEmbeddedIncluded(
+				"level1", rootContext, typeModel1Mock, "level1.",
+				null, null, excludePaths
+		);
+		checkFooBarExcluded( "", level1Context );
+		checkFooBarIndexedEmbeddedExcluded( level1Context, typeModel2Mock );
+		checkLeafExcluded( "level3", level1Context, "level3" );
+		checkCompositeExcluded( "level3", level1Context, "level3" );
+		checkLeafIncluded( "level2", level1Context, "level2" );
+		// Names including dots and matching the filter will be accepted
+		checkLeafIncluded( "level2.level3", level1Context, "level2.level3" );
+		checkCompositeIncluded( "level2.level3", level1Context, "level2.level3" );
+
+		// Check non-IndexedEmbedded nesting
+
+		IndexSchemaNestingContext level2NonIndexedEmbeddedContext =
+				checkCompositeIncluded( "level2", level1Context, "level2" );
+		checkFooBarExcluded( "", level2NonIndexedEmbeddedContext );
+		checkLeafIncluded( "level3", level2NonIndexedEmbeddedContext, "level3" );
+		checkLeafIncluded( "prefix2_level3", level2NonIndexedEmbeddedContext, "prefix2_level3" );
+		checkCompositeIncluded( "level3", level2NonIndexedEmbeddedContext, "level3" );
+		checkCompositeIncluded( "prefix2_level3", level2NonIndexedEmbeddedContext, "prefix2_level3" );
+		checkLeafExcluded( "level3.foo", level2NonIndexedEmbeddedContext, "level3.foo" );
+		checkCompositeExcluded( "level3.foo", level2NonIndexedEmbeddedContext, "level3.foo" );
+
+		// Check IndexedEmbedded composition without a prefix
+
+		ConfiguredIndexSchemaNestingContext level2Context = checkSimpleIndexedEmbeddedIncluded(
+				"level2", level1Context, typeModel2Mock, "level2.",
+				null, null, null
+		);
+		checkFooBarExcluded( "", level2Context );
+		checkFooBarIndexedEmbeddedExcluded( level2Context, typeModel3Mock );
+		checkLeafIncluded( "level3", level2Context, "level3" );
+		checkLeafIncluded( "prefix2_level3", level2Context, "prefix2_level3" );
+		checkCompositeIncluded( "level3", level2Context, "level3" );
+		checkCompositeIncluded( "prefix2_level3", level2Context, "prefix2_level3" );
+		checkLeafExcluded( "level3.foo", level2Context, "level3.foo" );
+		checkCompositeExcluded( "level3.foo", level2Context, "level3.foo" );
+
+		// Check IndexedEmbedded composition with a prefix
+
+		level2Context = checkSimpleIndexedEmbeddedIncluded(
+				"level2", level1Context, typeModel2Mock, "level2.prefix2_",
+				null, null, null
+		);
```

Comment:
I think the rest of the test are split into one test method per scenario... perhaps it would be good to do the same here? It's already complex enough without having to figure out which parts of the code above in the same method are relevant and which are not xD